<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>664</width>
    <height>433</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="title">
         <string>Исходные данные</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QPlainTextEdit" name="text_input">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="undoRedoEnabled">
            <bool>false</bool>
           </property>
           <property name="readOnly">
            <bool>true</bool>
           </property>
           <property name="textInteractionFlags">
            <set>Qt::TextSelectableByMouse</set>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <property name="sizeConstraint">
         <enum>QLayout::SetFixedSize</enum>
        </property>
        <item>
         <widget class="QGroupBox" name="groupBox_2">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="title">
           <string>Исходный файл</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_5">
           <item>
            <widget class="QLineEdit" name="text_fileName">
             <property name="text">
              <string/>
             </property>
             <property name="readOnly">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <item>
              <widget class="QPushButton" name="btn_openFile">
               <property name="text">
                <string>Выбрать файл</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btn_loadFile">
               <property name="text">
                <string>Загрузить файл</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_3">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="title">
           <string>Поиск идентификатора</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_6">
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_7">
             <property name="sizeConstraint">
              <enum>QLayout::SetFixedSize</enum>
             </property>
             <item>
              <layout class="QFormLayout" name="formLayout">
               <property name="sizeConstraint">
                <enum>QLayout::SetFixedSize</enum>
               </property>
               <property name="fieldGrowthPolicy">
                <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
               </property>
               <item row="0" column="0">
                <widget class="QLineEdit" name="le_identifier">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QPushButton" name="pushButton_3">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="text">
                  <string>Поиск</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="l_searchCount">
                 <property name="text">
                  <string>Всего поиск: 0 раз</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QPushButton" name="pushButton_4">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="text">
                  <string>Сброс</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <item>
              <widget class="QGroupBox" name="gb_ht_stats">
               <property name="title">
                <string>Хэш-таблица</string>
               </property>
               <widget class="QLabel" name="l_ht_avgCompares">
                <property name="geometry">
                 <rect>
                  <x>10</x>
                  <y>80</y>
                  <width>151</width>
                  <height>21</height>
                 </rect>
                </property>
                <property name="text">
                 <string>В среднем: 0</string>
                </property>
               </widget>
               <widget class="QLabel" name="l_ht_curCompares">
                <property name="geometry">
                 <rect>
                  <x>10</x>
                  <y>20</y>
                  <width>141</width>
                  <height>21</height>
                 </rect>
                </property>
                <property name="text">
                 <string>Сравнений: 0</string>
                </property>
               </widget>
               <widget class="QLabel" name="l_ht_totalCompares">
                <property name="geometry">
                 <rect>
                  <x>10</x>
                  <y>50</y>
                  <width>151</width>
                  <height>21</height>
                 </rect>
                </property>
                <property name="text">
                 <string>Всего сравнений: 0</string>
                </property>
               </widget>
              </widget>
             </item>
             <item>
              <widget class="QGroupBox" name="gb_sl_stats">
               <property name="title">
                <string>Упорядоченный список</string>
               </property>
               <widget class="QLabel" name="l_sl_curCompares">
                <property name="geometry">
                 <rect>
                  <x>10</x>
                  <y>20</y>
                  <width>151</width>
                  <height>16</height>
                 </rect>
                </property>
                <property name="text">
                 <string>Сравнений: 0</string>
                </property>
               </widget>
               <widget class="QLabel" name="l_sl_totalCompares">
                <property name="geometry">
                 <rect>
                  <x>10</x>
                  <y>50</y>
                  <width>151</width>
                  <height>21</height>
                 </rect>
                </property>
                <property name="text">
                 <string>Всего сравнений: 0</string>
                </property>
               </widget>
               <widget class="QLabel" name="l_sl_avgCompares">
                <property name="geometry">
                 <rect>
                  <x>10</x>
                  <y>80</y>
                  <width>141</width>
                  <height>21</height>
                 </rect>
                </property>
                <property name="text">
                 <string>В среднем: 0</string>
                </property>
               </widget>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_5">
          <property name="text">
           <string>Выход из программы</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>664</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>pushButton_5</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>468</x>
     <y>411</y>
    </hint>
    <hint type="destinationlabel">
     <x>325</x>
     <y>216</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
